tools/mock-node/src/lib.rs

```diff
@@ -457,15 +444,88 @@ impl MockPeer {
     }
 }
 
+// The mock node accepts from `listener` and starts a new MockPeer to handle
+// each incoming connection
+struct MockNode {
+    listener: Listener,
+    chain: ChainStoreAdapter,
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    network_start_height: BlockHeight,
+    network_config: MockNetworkConfig,
+    block_production_delay: Duration,
+}
+
+impl MockNode {
+    async fn new(
+        chain: ChainStoreAdapter,
+        epoch_manager: Arc<dyn EpochManagerAdapter>,
+        genesis_hash: CryptoHash,
+        secret_key: SecretKey,
+        listen_addr: tcp::ListenerAddr,
+        chain_id: String,
+        archival: bool,
+        block_production_delay: Duration,
+        shard_layout: ShardLayout,
+        network_start_height: BlockHeight,
+        network_config: MockNetworkConfig,
+    ) -> anyhow::Result<Self> {
+        let listener = Listener::bind(
+            listen_addr,
+            secret_key,
+            &chain_id,
+            genesis_hash,
+            network_start_height,
+            shard_layout.shard_ids().collect(),
+            archival,
+            30 * near_time::Duration::SECOND,
+        )
+        .await?;
+
+        Ok(Self {
+            listener,
+            chain,
+            epoch_manager,
+            network_start_height,
+            network_config,
+            block_production_delay,
+        })
+    }
+
+    // listen on the addr passed to MockPeer::new() and wait til someone connects.
```

Comment:
```suggestion
    // listen on the addr passed to MockPeer::new() and wait till someone connects.
```

where is spellcheck...

Reply:
lol I think I've seen both spellings as acceptable, but how about just "until"